packages/media/src/getBlurFromImage.js

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { encode } from 'blurhash';
+/**
+ * Internal dependencies
+ */
+import preloadImage from './preloadImage';
+
+const getImageData = (image) => {
+  const { width, height } = image;
+  const canvas = document.createElement('canvas');
+  canvas.width = width;
+  canvas.height = height;
+
+  const ctx = canvas.getContext('2d');
+  ctx.drawImage(image, 0, 0);
+  return ctx.getImageData(0, 0, width, height);
+};
+
+const encodeImageToBlurhash = async (src) => {
+  const image = await preloadImage(src);
+  const imageData = getImageData(image);
+  const { data, width, height } = imageData;
+  // todo workout why 1 and 1 the only options that are not painfully slow.
+  return encode(data, width, height, 4, 4);
```

Comment:
@swissspidy 4, 4 are the default values here. However, using these values, the editor is unable. If you change to 9,9, same story. 1,1 is usable, but does not generate a valid blurhash.

Reply:
The `useDetectBlurHash` hook calls this function with the full image size (`resource.src`).

Can we try to use a smaller thumbnail size instead. Can use `getSmallestUrlForWidth(0, resource)` for that.

Ideally with some quick time tracking to know how long it takes.

If that doesn't help, we can explore other options as discussed (not implementing, or doing it server-side)

Reply:
As discussed, another option would be to use a web worker (fairly easy to do with [worker-loader](https://v4.webpack.js.org/loaders/worker-loader/)), but we'd need to measure performance to compare all possible options

Reply:
In my testing, I managed to the speed to much faster.

Reply:
What made it faster?

Do you have any numbers?

Reply:
My tweaks to make it generate only one at a time. Which is not here. Also no numbers, the different between, editor completely locked and working fine.

Reply:
How fast is the blurhash generation _now_?

Reply:
Now that we are using smaller images too, the average blurhash is around 100-150ms, over 1000 ms of full sized resources. Not to mention the need to down the full size image, which is wasteful. But poster images could be pretty large, which is why we limit to one being processed at a time.

Reply:
Could you add `getTimeTracker` here?

Reply:
I have a tracker elsewhere, want me to move it here?